File: example_services/java/finatra/example/example-service/src/main/java/com/me/example/handlers/GetMessageHandler.java
package com.me.example.handlers;

import com.me.example.persistence.MessageDao;
import com.twitter.finagle.http.Request;
import com.twitter.finagle.http.Response;
import com.twitter.finagle.http.Status;
import com.twitter.finatra.http.exceptions.HttpException;
import com.twitter.finatra.http.exceptions.NotFoundException;
import com.twitter.finatra.http.response.ResponseBuilder;
import org.jboss.netty.handler.codec.http.HttpResponseStatus;

public class GetMessageHandler implements RequestHandler {
    private final MessageDao messageDao;
    public GetMessageHandler(MessageDao messageDao) {
        this.messageDao = messageDao;
    }

    @Override
    public Object handle(Request request) {
        return messageDao.get(request.getParam("id")).orElse(null);
    }
}

File: example_services/scala/finatra/example/example-service/src/main/scala/com/me/example/handlers/GetMessageHandler.scala
package com.me.example.handlers

import javax.inject.Inject

import com.me.example.domain.http.GetMessageRequest
import com.me.example.persistence.MessageDao
import com.twitter.util.Future

class GetMessageHandler @Inject()(
  dao: MessageDao,
  request: GetMessageRequest
) extends RequestHandler {

  override def handle(): Future[Any] = {
    dao.get(request.id)
  }
}